resources/newssite/news-nuxt/context/DataContext.vue

```diff
@@ -0,0 +1,42 @@
+<script>
+import { provide } from "vue";
```

Comment:
this should be refactored as a composable rather than a context provider, which is much more a react pattern

the content of the script here can be extracted into a file `~/composables/provide-locale.ts` with an export `export function provideLocale () {/* */}`. Then this function can be run directly in the `setup()` function of `app.vue`.

Reply:
done! so much nicer